service/src/test/java/bio/terra/workspace/common/utils/MockMvcUtils.java

```diff
@@ -910,14 +909,44 @@ public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucket(
   }
 
   /** Call cloneGcsBucket() and return immediately; don't wait for flight to finish. */
-  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsync(
+  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketWithError(
       AuthenticatedUserRequest userRequest,
       UUID sourceWorkspaceId,
       UUID sourceResourceId,
       UUID destWorkspaceId,
       ApiCloningInstructionsEnum cloningInstructions,
       String destBucketName,
-      int expectedCode)
+      int statusCode)
+      throws Exception {
+    ApiCloneControlledGcpGcsBucketRequest request =
+        new ApiCloneControlledGcpGcsBucketRequest()
+            .destinationWorkspaceId(destWorkspaceId)
+            .cloningInstructions(cloningInstructions)
+            .name(TestUtils.appendRandomNumber(DEST_BUCKET_RESOURCE_NAME))
+            .jobControl(new ApiJobControl().id(UUID.randomUUID().toString()));
+    if (destBucketName != "") {
+      request.bucketName(destBucketName);
+    }
+    mockMvc
+        .perform(
+            addJsonContentType(
+                addAuth(
+                    post(CLONE_CONTROLLED_GCP_GCS_BUCKET_FORMAT.formatted(
+                            sourceWorkspaceId, sourceResourceId))
+                        .content(objectMapper.writeValueAsString(request)),
+                    userRequest)))
+        .andExpect(status().is(statusCode));
+    // If an exception was thrown, deserialization won't work, so don't attempt it.
+    return null;
+  }
+
+  public ApiCloneControlledGcpGcsBucketResult cloneControlledGcsBucketAsyncSuccessfully(
```

Comment:
// This checks 200 and 204. (For some tests, sometimes the test will return 200 and other times 204.)